fences/agent.py

```diff
@@ -32,73 +39,131 @@
 """
 
 class Agent():
-  def __init__(self, api_key):
-    self.initialize_agent(api_key)
+    def __init__(self):
+        self.initialize_agent()
 
-  def initialize_agent(self, api_key):
-    self.client = anthropic.Anthropic(api_key=api_key)
+    def initialize_agent(self):
+        api_choice = input("Do you want to use OpenAI or Anthropic? (openai/anthropic): ").lower()
```

Comment:
I want to keep this outside of the agent so all the interactions with the command line are organised under the cli.py. This way agent.py can be used as a standalone library in the future.

Can you move all the inputs and prints to console back to an argument to the constructor?

Reply:
Sure, it makes all sense, just commited it!